src/components/back-link-helper/back-link-helper.js

```diff
@@ -14,14 +14,27 @@ BackLinkHelper.prototype.init = function init() {
        Currently, a page refresh sets the referer to empty, leading to the back link being hidden
        under our existing logic.
      */
-    // eslint-disable-next-line max-len
-    const referrerNotOnSameDomain = () => {
-      const referer = this.document.referrer;
-      return !referer || referer.indexOf(this.window.location.host) === -1;
+    const invalidDomain = () => {
+      const ref = this.document.referrer;
+      let isInvalid = false;
+
+      if (ref) {
+        const diffDom = ref.indexOf(this.window.location.host) === -1;
+        if (diffDom) {
+          // Allow PEGA domains
+          const allowList = ['account-np.hmrc.gov.uk', 'account.hmrc.gov.uk'];
+          const allowed = allowList.some((e) => ref.includes(e));
+          isInvalid = !allowed;
+        }
+      } else {
+        isInvalid = true;
+      }
+
+      return isInvalid;
```

Comment:
we can simplify the logic here with an early return, and I think invalidDomain doesn't tell us much about the intent of what we're checking, how about something like this:
```suggestion
    const referrerAllowList = ['account-np.hmrc.gov.uk', 'account.hmrc.gov.uk'];
    const shouldHideBackLink = () => {
      const referrer = this.document.referrer;
      if (!referrer) return true
      const referredFromSameDomain = referrer.indexOf(this.window.location.host) >= 0;
      const referrerOnAllowList = referrerAllowList.some((allowListedDomain) => referrer.includes(allowListedDomain));
      return referredFromSameDomain || referrerOnAllowList
```
I don't think we're saving much by shortening variable names, it's more readable to use the full words - for example, I think "ref" and "diffDom" are a bit ambiguous

Reply:
one gap this highlights for me too, is that the current implementation would be so easy to circumvent by putting the current url in a query param on the referring url

Reply:
I think it's good that we're removing it in the future